account_analytic_parent/models/account_analytic_account.py

```diff
@@ -27,10 +27,25 @@ def _compute_debit_credit_balance(self):
         of analytic account changes
         """
         super(AccountAnalyticAccount, self)._compute_debit_credit_balance()
-        for account in self:
-            account.debit += sum(account.mapped('child_ids.debit'))
-            account.credit += sum(account.mapped('child_ids.credit'))
-            account.balance += sum(account.mapped('child_ids.balance'))
+        analytic_line_obj = self.env['account.analytic.line']
+        # compute only analytic line
+        for account in self.filtered(lambda x: x.child_ids):
+            domain = [('account_id', 'child_of', account.ids)]
+            credit_groups = analytic_line_obj.read_group(
+                domain=domain + [('amount', '>=', 0.0)],
```

Comment:
No need to include 0 results

Reply:
I mean they are not going to add anything to the sum, as their amount is 0, so better to strip them.

Reply:
yes, it makes sense